File: yugioh-20221012/src/main/java/tw/gaas/yugioh/domain/card/Cards.java
package tw.gaas.yugioh.domain.card;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cards {

    // 上限
    private Integer limit;
    // 元素
    private List<Card> elements;
}

File: yugioh-kehao/framework/src/main/kotlin/tw/gaas/yugioh/model/dto/Cards.kt
package tw.gaas.yugioh.model.dto

import io.swagger.v3.oas.annotations.media.Schema
import lombok.Data
import org.springframework.validation.annotation.Validated

/**
 * 卡片集合，目前採取 array 形式存在
 */
@Schema(description = "卡片集合")
@Validated
@Data
class Cards : ArrayList<Card?>() {
}
